api/qanpb/filters.proto

```diff
@@ -0,0 +1,41 @@
+syntax = "proto3";
+
+package qan;
+
+option go_package = "qanpb";
+
+import "google/api/annotations.proto";
+import "google/protobuf/timestamp.proto";
+
+// Filters serves list of names of metrics.
+service Filters {
+  // GetFilters gets map of metrics names.
+  rpc GetFilters(FiltersRequest) returns (FiltersReply) {
+    option (google.api.http) = {
+      post: "/v1/qan/GetFilters"
+      body: "*"
+    };
+  }
+}
+
+// FiltersRequest contains period for which we need filters.
+message FiltersRequest {
+  google.protobuf.Timestamp from = 1;
+  google.protobuf.Timestamp to = 2;
+}
+
+// FiltersReply is map of labels for given period by key.
+message FiltersReply {
+  map<string, ListLabels> labels = 1;
+}
+
+// ListLabels is list of label values.
+message ListLabels {
+  repeated NameAndCount values = 1;
```

Comment:
Can it contain duplicate names? If not, can it be map too?

Reply:
No, it cannot contains duplicates. But I do not see much benefits in smaller response versus more complex code: 5 additional loops into Go code to convert data from DB rows and on UI need an additional workaround to iterate over the map (https://github.com/angular/angular/issues/2246)

Reply:
Then we need to document the fact that duplicates are not possible.